src/main/typescript/data/item/magic/properties.ts

```diff
@@ -0,0 +1,26 @@
+import { MagicType, getMagicType } from "../../../constants.js";
+import type WvItem from "../../../item/wvItem.js";
+import BaseItemProperties from "../common/baseItem/properties.js";
+import RulesProperties from "../common/rules/properties.js";
+import type MagicDataSource from "./source.js";
+import { MagicDataSourceData } from "./source.js";
+
+export default interface MagicDataProperties extends MagicDataSource {
+  data: MagicDataPropertiesData;
+}
+
+export class MagicDataPropertiesData
+  extends MagicDataSourceData
+  implements BaseItemProperties
+{
+  constructor(source: MagicDataSourceData, owningItem: WvItem) {
+    super();
+    foundry.utils.mergeObject(this, source);
+    this.type = getMagicType(this.school);
+    BaseItemProperties.transform(this, source, owningItem);
+  }
+
+  override rules = new RulesProperties();
+
+  type: MagicType = getMagicType(this.school);
```

Comment:
If you put this in the constructor, remove the default value here.

Reply:
Oh, my bad. I thought the typechecker would insist that I initialise it, but apparently it's smart enough to notice initialisation happens in the ctor. Fixed.